docs/src/SearchField.doc.js

```diff
@@ -68,6 +68,15 @@ card(
         name: 'value',
         type: 'string',
       },
+      {
+        name: 'errorMessage',
```

Comment:
@szatakia could you explain (perhaps share a screenshot) why we need both `hasError` and `errorMessage`? It's confusing to have both, e.g. what should happen when `hasError` is `false` and `errorMessage` is set to a string?

For now I would suggest to only keep `errorMessage` and remove `hasError`:
* It makes it clear to the user why we show an error.
* Consistent with other text inputs: `TextField` / `TextArea`

Reply:
Hi @christianvuerings I originally suggestedincluding `hasError` for parity with `TextField` since it's here: https://github.com/pinterest/gestalt/blob/master/packages/gestalt/src/TextField.js#L21

However, I noticed it does not appear on Gestalt Docs so I wonder if we really need them all. Do you have any thoughts on that?

To me it seems it's a way to highlight the field with error border when there is no message to be displayed? Not really sure

Reply:
@themindfuldev you are right - it's in the code but not documented. Let's remove `hasError` from `SearchField` for now.

I looked at the stats for the Pinterest codebase and we only have 6 instances of `hasError` compared to 154 instances of `errorMessage` on `TextField`.

![image](https://user-images.githubusercontent.com/127199/91107370-ff548c00-e629-11ea-8a88-0bd96ccdc016.png)

Reply:
I did that just to make it consistent with error-handling capabilities of [TextField](https://github.com/pinterest/gestalt/blob/954da26894743e2d6abf5be85125c34b1743728a/packages/gestalt/src/TextField.js#L21). But it makes sense. I will simplify this and will create a PR for TextField too.

Reply:
@szatakia thank you! For `TextField`, we should first update the 6 places in our Pinterest codebase where we still use `hasError` and replace those with `errorMessage`.

Reply:
Removed `hasError` from SearchField